.travis.yml

```
@@ -19,3 +21,15 @@ notifications:
 after_success:
   # push coverage results to Coveralls
   - julia -e 'using Pkg; Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())'
+
+jobs:
+  include:
+    # Make sure ChainRules.jl works with this
+    - state: Integration
+      name: ChainRules
+      julia: 1.0
+      os: linux
+      script:
+        - git clone "https://github.com/JuliaDiff/ChainRules.jl.git"
+        - julia --project="ChainRules.jl" -e "using Pkg; Pkg.pkg\"dev .\"; Pkg.pkg\"up\"; Pkg.pkg\"test\""
```

this is unfortunately difficult to read (for me) ...it's sad if Pkg doesn't have a nicer way to write this (which it might not)

---

I hope this is equivalent 
```suggestion
        - julia --project="ChainRules.jl" -e "using Pkg; Pkg.develop("."); Pkg.update(); Pkg.test()"
```

---

```suggestion
        # Start julia with project set to newly clones subdir ChainRules.jl
        # dev the current directory to get the version of ChainRulesCore that we are testing
        # update everything to resolve and sortout any versions,
        # finally test the ChainRules.jl
        - julia --project="ChainRules.jl" -e "using Pkg; Pkg.pkg\"dev .\"; Pkg.pkg\"up\"; Pkg.pkg\"test\""
```

---

i like both these changes :)

---

> I hope this is equivalent

I was wrong. I am sorry. pkg_str is the only human-friendly way to interact with Pkg.